### DISCUSSION TITLE
Not able to pull items from a chest to a turtle using peripheral

### DISCUSSION THREAD

**Discussion by iridiumlynx** [cc-tweaked/CC-Tweaked#601] [Q&A]
I can't transfer items from/to a wrapped chest peripheral using pullItem() or pushItem(). These methods just move items in the same inventory, without a possibility to transfer items between multiple inventories. If it was an intended behavior, please make a method of transferring items between _specific_ slots of _multiple_ inventories (including turtle's inventory).

 - Minecraft version: 1.16.4
 - CC: Tweaked version: 1.94.0
 - Logs: there is no runtime exception, just weird behavior.

Detailed reproduction steps: 
  1. Wrap a storage block, i.e. a chest, as a peripheral
  2. Execute pullItem() on any slots in that storage block.
  3. Watch the item being transferred from a slot in inventory to another slot in the same inventory instead of being transferred to the turtle.

**Comment by SquidDev** [comment]
Can you post the code you're running, as well as a screenshot of your setup? I tested this fairly recently, so am pretty sure it works.

**Reply by pipe01** [reply]
Could you send some code to pull items from a chest into a turtle? I can't figure out how to make the turtle the target of the pushItems method.

**Reply by fatboychummy** [reply]
You cannot transfer directly to the turtle without a block modem, but assuming one is in front of the turtle with a chest connected to it:

```lua
local modem = peripheral.wrap("front")
local turtleName = modem.getNameLocal()
local chest = peripheral.find("chest")

chest.pushItems(turtleName, 1, 1)
```

Will push 1 item from slot 1 of the chest to the turtle.

Edit for future clarification: Make sure you right-click the modem so it says in chat "Peripheral chest_xyz connected to network", and the modem has the little red ring on it.

**Reply by pipe01** [reply]
Ah got it, thank you.

**Reply by earlPeanutButter** [reply]
Surprisingly difficult to find the answer to this. Thanks fatboy!

**Comment by Lupus590** [comment]
@SquidDev I think they are trying to have the turtle be the target of the generic peripheral method. I don't think you added support for that.

**Comment by SnazzGass** [comment]
yes it seems like turtles don't have the generic peripheral inventory

**Reply by Wojbie** [reply]
Turtles don't have generic peripheral cause they have a specific peripheral already set on them. They can still be used as target for generic peripheral inventory.. you just need to call functions on other side of conversation like described in [this ](https://github.com/cc-tweaked/CC-Tweaked/discussions/601#discussioncomment-1046232) message above. Chest is moving items to/from turtle.
